### ISSUE TITLE
sequence diagram request: grouping actors in box

### ISSUE THREAD

**Issue by rusko995** [terrastruct/d2#1320]
I'd like to propose the same thing as it was requested for Mermaid in [this issue](https://github.com/mermaid-js/mermaid/issues/1505).

I'd love to be able to do this:
![image](https://github.com/terrastruct/d2/assets/22886512/33b68139-37df-4def-a826-5ab342074278)

More specifically, the "boxes" labeled "Application" and "Kafka".

Proposed syntax:
```
shape: sequence_diagram
box Application: {
  producer
  consumer
}
box Kafka: {
  tc: transaction coordinator
  sourceTopic
  targetTopic
}
producer -> consumer: initTransactions
producer -> producer: beginTransactions
consumer -> sourceTopic: poll messages
...
```
In the box there should be an option to define participants with simple definition or with alias (e.g. `tc: transaction coordinator`)

**Comment by soma-m** [comment]
I wondering if there is a simpler and more generic way to do this without introducing a new box keyword.

**Comment by rusko995** [comment]
Maybe if the "box" keyword is left-out? 

But yes, at the end you want to have a box and you want to name it somehow. Then it is good to bound the area of the box with the brackets {}.

**Comment by alixander** [comment]
yeah this should be doable with the `box` keyword left out. i consider this a todo

**Comment by kskofenko-dsr** [comment]
Any updates on the request?

**Comment by bo-ku-ra** [comment]
forced

https://play.d2lang.com/?script=nE9BbgMhDLxH6h_8AVY9g7RfqcyuaVAJTg1VQ6L8vTJko_RQReplMLZnPIOXlx1A2eORLBT6_KK80Nsa8V3w4HSGCl5hUeiFmdEG5u2PE15KbYmmUoU_yHzHte4tvLrRDTElC1UwlyMK5equD1QwRh_7h8T1vjb3NY-ykfvIP01wUmgK515NuAm0_xlvw3R7Yvpk5vPd5i3FyOEn5QZmj3JTCJyrWTixWBBaHTwK_2oNU4dYahMuNG7tfgIAAP__&

**Comment by kskofenko-dsr** [comment]
@bo-ku-ra This doesn't look like a maintainable approach. The arrow is always linked to the middle of the method, moreover it sticks together with the response arrow. + Nested sequence diagrams have different height 

![image](https://github.com/user-attachments/assets/f00afb17-359d-40f2-9b03-76e2ddee9dbd)
https://play.d2lang.com/?script=hI1LCgIxEET3dYq6gDlAFl5FOrEZA5rEdARHmbvLDKIBF-768-qVeD5B2kmqeppeb5qjHo5JpiYXkAIygIxYEP7Bd5AzyAcWQJw44W7PbfBMOfUkZ9ZWopqlPH2R4OYVaavSOrZ1uA-Rz-ttbWo1F9Ofulqsj8FXAAAA__8%3D&
